### DISCUSSION TITLE
Querying $in from a list

### DISCUSSION THREAD

**Discussion by zzph** [deta/cloud-docs#605] [Help]
Hi,

How can I fetch based on multiple parameters of the same key? Similar to an `$in` query. I tried the follow:

`Deta.Base("foo").fetch({key:['bar','baz']})`

`Deta.Base("foo").fetch({"key?in":['bar','baz']})`

`Deta.Base("foo").fetch([{key:'bar'}, {key:'baz'}] )`

So the above should return *any* items which match *any* of the `key` values

Following your docs, I tried both of those. They both error wit error: `unhandledRejection: Error: Bad query`.

PS- Deta looks incredible and simple. Docs and UI are all fantastic too- well done.

**Comment by lemonyte** [comment]
From the [docs](https://docs.deta.sh/docs/base/queries#querying-keys), AND & OR queries for keys are not supported. Instead, you could get each key individually, and add them to a list if an item was found.

**Reply by zzph** [reply]
Thanks. Two quick questions:

1. Are you suggesting to do something like this? Fetch all items and do the filter after?
`Deta.Base("foo").fetch().then(({items})=>items.filter(i=>filterKeys.includes(i.key)))`

2. Does that mean, renaming the field `key` field to something else would work? EG if I created my own `id` field for example

**Comment by jnsougata** [comment]
> Hi,
> 
> How can I fetch based on multiple parameters of the same key? Similar to an `$in` query. I tried the follow:
> 
> `Deta.Base("foo").fetch({key:['bar','baz']})`
> 
> `Deta.Base("foo").fetch({"key?in":['bar','baz']})`
> 
> `Deta.Base("foo").fetch([{key:'bar'}, {key:'baz'}] )`
> 
> So the above should return _any_ items which match _any_ of the `key` values
> 
> Following your docs, I tried both of those. They both error wit error: `unhandledRejection: Error: Bad query`.
> 
> PS- Deta looks incredible and simple. Docs and UI are all fantastic too- well done.

if you meant `key` as `field` and want to do membership check w.r.t a list of strings, you can look into it: https://docs.deta.sh/docs/base/queries/#contains 
![image](https://user-images.githubusercontent.com/53375272/203658261-593853e1-0a35-47e7-92d5-3cf1215f3537.png)

**Reply by zzph** [reply]
No, the `?contains` operator does not help. I want to check if it contains any of my search options I pass it, eg:

`
Deta.Base("dogs").fetch({owner:['bar','baz']})
`

Which should only return `dogs` which have an owner of `foo` or `baz` eg:

`
[{key:'x', name:'fido', owner:'foo'}, {key:'y', name:'biff', owner:'foo'}. {key:'z', name:'rod', owner:'baz'}]
`

Is that possible?

**Reply by lemonyte** [reply]
Yes, you can use OR'ed queries on fields other than `key`, i.e., `[{"owner": "bar"}, {"owner": "baz"}]`. [Docs reference](https://docs.deta.sh/docs/base/queries#or).
